src/test/java/com/pusher/client/channel/impl/ChannelImplTest.java

```diff
@@ -27,7 +29,14 @@
 
     @Before
     public void setUp() {
-        when(factory.getEventQueue()).thenReturn(new InstantExecutor());
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                final Runnable r = (Runnable) invocation.getArguments()[0];
+                new InstantExecutor().execute(r);
```

Comment:
This `InstantExecutor` class is probably redundant now.

`((Runnable)invocation.getArguments()[0]).run();` would do the trick.

To stick to proper OO patterns, we should really create a `GloballySynchronizedExecutor` which takes the lock object as a constructor arg, then we could still return a dummy type here.
Not going to insist on it though.